### ISSUE TITLE
Access token generation from refresh token fails.

### ISSUE THREAD

**Issue by Saubhy** [ballerina-platform/ballerina-lang#10608]
**Description:**
There is a Salesforce connector and a endpoint implemented. All the requests to retrieve data from the Salesforce endpoint are working if the access token is valid. After the access token expiration, it is not being regenerated using the request token. 

**Steps to reproduce:**

> endpoint sfdc:Client salesforceClientEP {
    clientConfig: {
        url: config:getAsString("SALESFORCE_ENDPOINT"),
        auth: {
            scheme: http:OAUTH2,
            accessToken: config:getAsString("SALESFORCE_ACCESS_TOKEN"),
            refreshToken: config:getAsString("SALESFORCE_REFRESH_TOKEN"),
            clientId: config:getAsString("SALESFORCE_CLIENT_ID"),
            clientSecret: config:getAsString("SALESFORCE_CLIENT_SECRET"),
            refreshUrl: config:getAsString("SALESFORCE_REFRESH_URL")
        }
    }
};

> function fetchSalesforceData(string|json jiraKeysOrNextRecordUrl) returns json|sfdc:SalesforceConnectorError {
    match jiraKeysOrNextRecordUrl {
        string nextRecordUrl => {
            var connectorResponse = salesforceClientEP->getNextQueryResult(nextRecordUrl);
            return connectorResponse;
        }
        json jiraKeys => {
            string SOQuery = buildQueryFromTemplate(QUERY_TEMPLATE_GET_SALESFORCE_DATA_BY_JIRA_KEY, jiraKeys);
            var connectorResponse = salesforceClientEP->getQueryResult(SOQuery);
            return connectorResponse;
        }
    }
}

Error:
> 2018-09-28 14:08:50,441 ERROR [wso2/dbSync:0.1.0] - No data returned from salesforce API : {message:"Http error -> message: Failed to generate new access token from the given refresh token", cause:{message:"", cause:null}}

**Affected Versions:**
0.980.0

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by NipunaRanasinghe** [comment]
This can be observed if either the refresh token or the refresh URL is invalid. Therefore can you please verify that both refresh token and refresh URL are valid?

**Comment by Saubhy** [comment]
Yes, both of them are valid. I can generate the access token by using the same refresh token and refresh url through a curl command.

**Comment by Anoukh** [comment]
@Saubhy seems like the salesforce [API](https://help.salesforce.com/articleView?id=remoteaccess_oauth_refresh_token_flow.htm&type=0) used to generate the access token is different to our implementation. We will look into it further and get back to you.
